SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+        file = os.path.join(path, filename)
+        zip_obj.write(file, outdir + "/" + os.path.basename(file))
+        if os.path.isdir(file):
+            zip_add_folder(zip_obj, file, outdir + "/" + filename)
+
+def make_zip(board_name, path):
+    zip_obj = ZipFile(path + "/" + board_name + ".zip", 'w')
+    zip_dir = path + "/" + SQUARELINE_OUTDIR
+    zip_add_folder(zip_obj, zip_dir, SQUARELINE_OUTDIR)
+    zip_obj.close()
+
+def check_json_key(manifest, key):
+    if key not in manifest:
+        print_error(f"Missing {key} in manifest JSON file")
+        os._exit(1)
+
+def get_file_version(manifest):
+    check_json_key(manifest, "version")
+    v = manifest["version"].split('.')
+    return "_v" + v[0] + "_" + v[1] + "_" + v[2]
+
+def create_slb_file(output, output_filename, manifest):
+    check_json_key(manifest, "name")
+    check_json_key(manifest, "version")
+    check_json_key(manifest, "mcu")
+    check_json_key(manifest, "screen_width")
+    check_json_key(manifest, "screen_height")
+    check_json_key(manifest, "screen_color_swap")
+    check_json_key(manifest, "short_description")
+    check_json_key(manifest, "long_description")
+
+    SLB_FILE["version"] = manifest["version"]
+    SLB_FILE["title"] = manifest["name"]
+    SLB_FILE["keywords"] += ", " + manifest["name"] + ", " + manifest["mcu"]
+    SLB_FILE["width"] = manifest["screen_width"]
+    SLB_FILE["width_min"] = manifest["screen_width"]
+    SLB_FILE["width_max"] = manifest["screen_width"]
+    SLB_FILE["height"] = manifest["screen_height"]
+    SLB_FILE["height_min"] = manifest["screen_height"]
+    SLB_FILE["height_max"] = manifest["screen_height"]
+    if manifest["screen_color_swap"]:
+        SLB_FILE["color_depth"] = "16 sw"
+    else:
+        SLB_FILE["color_depth"] = "16"
+    SLB_FILE["short_description"] = manifest["short_description"]
+    SLB_FILE["long_description"] = manifest["long_description"]
+
+    out_slb_json = json.dumps(SLB_FILE, indent=4)
+    f = open(output + "/" + output_filename + ".slb", "w")
+    f.write(out_slb_json)
+    f.close()
+
+    print(f"  File {output_filename}.slb created.")
+
+
+# Process of the board generating
+def process_board(board_name, output, dir):
+    global placeholders
+    print(f"Processing board: {dir}")
+
+    path = dir + "/" + BOARD_GENFILE
+    if os.path.exists(path):
+        # Parse manifest file
+        f = open(path, "r")
+        manifest = json.loads(f.read())
+        f.close()
+
+        # Get string version for file names
+        file_ver = get_file_version(manifest)
+        output_filename = board_name + file_ver
+
+        check_json_key(manifest, "placeholders")
+        placeholders = manifest["placeholders"]
+
+        # Copy common files
+        copy_directory(COMMON_DIR, output + "/" + SQUARELINE_OUTDIR)
+        # Copy specific board files
+        for file in BOARD_SPECIFIC_FILES:
+            copy_file(dir + "/" + file, output + "/" + SQUARELINE_OUTDIR + "/" + file)
```

Comment:
Suggest using `os.path.join` for building paths. It will use correct type of path separator for each OS.